gateway/reverse_proxy_test.go

```diff
@@ -181,7 +181,9 @@ func TestReverseProxyDnsCache(t *testing.T) {
 	)
 
 	ts := StartTest(nil)
+	ts.MockHandle, _ = test.InitDNSMock(etcHostsMap, nil)
 	defer ts.Close()
+	defer ts.MockHandle.ShutdownDnsMock()
```

Comment:
If you see dnsmock being used somewhere and you're touching that part of the code, just remove it. There's no safe way to use the dns mock (particularly modifying net.DefaultResolver), and we have various utilities that can avoid it in the /test pkg (net.dialer, http client, etc.). Can this be avoided?